compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp

```diff
@@ -149,12 +147,17 @@ int32_t nchw_axis_to_nhwc(int32_t axis)
 
 luci::CircleTranspose *create_post_transpose(luci::CircleNode *node)
 {
-  return create_4d_transpose(node, {0, 3, 1, 2});
+  return create_Nd_transpose(node, {0, 3, 1, 2});
 }
 
 luci::CircleTranspose *create_pre_transpose(luci::CircleNode *node)
 {
-  return create_4d_transpose(node, {0, 2, 3, 1});
+  return create_Nd_transpose(node, {0, 2, 3, 1});
+}
+
+luci::CircleTranspose *create_post_transpose_3d(luci::CircleNode *node)
+{
+  return create_Nd_transpose(node, {0, 2, 1});
```

Comment:
Does {0, 2, 1} work for all cases?

Let's assume reduction_indices = 0 (N).

For NCHW (before transformation), the output will be CHW
For NHWC, the output will be HWC, so 3d transpose should be {2, 0, 1}.

Reply:
Ah, I completely missed the possibility that N axis can be reduced, let me rethink this PR